scaleway/config.go

```diff
@@ -3,36 +3,175 @@ package scaleway
 import (
 	"bytes"
 	"context"
+	"encoding/json"
+	"fmt"
 	"io"
 	"io/ioutil"
 	"log"
 	"net/http"
 	"os"
-	"sort"
 	"time"
 
+	"github.com/hashicorp/terraform/terraform"
+
+	"github.com/scaleway/scaleway-sdk-go/scw"
+
 	retryablehttp "github.com/hashicorp/go-retryablehttp"
 	"github.com/hashicorp/terraform/helper/logging"
-	sdk "github.com/nicolai86/scaleway-sdk"
+	deprecatedSdk "github.com/nicolai86/scaleway-sdk"
+	"github.com/scaleway/scaleway-sdk-go/logger"
+	"github.com/scaleway/scaleway-sdk-go/scwconfig"
+	"github.com/scaleway/scaleway-sdk-go/utils"
 )
 
-// Config contains scaleway configuration values
+// scwConfig contains the Scaleway configuration.
+var scwConfig scwconfig.Config
+
+func init() {
+	// Init the Scaleway config.
+	config, err := scwconfig.Load()
+	if err != nil {
+		log.Fatalf("error: cannot load configuration: %s", err)
+	}
+	scwConfig = config
+
+	// Init the SDK logger.
+	logger.SetLogger(sdkLogger{})
+}
+
+// Config is a configuration for a client.
 type Config struct {
-	Organization string
-	APIKey       string
-	Region       string
+	AccessKey             string
+	SecretKey             string
+	DefaultOrganizationID string
+	DefaultRegion         utils.Region
+	DefaultZone           utils.Zone
 }
 
-// Client contains scaleway api clients
-type Client struct {
-	scaleway *sdk.API
+// Meta contains SDK clients used by resources.
+//
+// This meta value is passed into all resources.
+type Meta struct {
+	// The Scaleway SDK client.
+	scwClient *scw.Client
+
+	// Deprecated: The deprecated Scaleway SDK (will be removed in `v2.0.0`).
+	deprecatedClient *deprecatedSdk.API
+}
+
+// Meta creates a meta instance from a client configuration.
+func (c *Config) Meta() (*Meta, error) {
+	meta := &Meta{}
+
+	// Scaleway Client
+	client, err := c.GetScwClient()
+	if err != nil {
+		return nil, err
+	}
+	meta.scwClient = client
+
+	// Deprecated Scaleway Client
+	deprecatedClient, err := c.GetDeprecatedClient()
+	if err != nil {
+		return nil, fmt.Errorf("error: cannot create deprecated client: %s", err)
+	}
+	meta.deprecatedClient = deprecatedClient
+
+	// Fetch known scaleway server types to support validation in r/server
+	err = fetchServerAvailabilities(client)
+	if err != nil {
+		log.Printf("error: cannot fetch server availabilities: %s", err)
+	}
+
+	return meta, nil
 }
 
-// client is a bridge between sdk.HTTPClient interface and retryablehttp.Client
+var userAgent = fmt.Sprintf("terraform-provider/%s;terraform/%s", version, terraform.VersionString())
```

Comment:
Is this user agent format documented somewhere?

Reply:
we should use space as separator `terraform-provider/%s terraform/%s`